src/commands.rs

```diff
@@ -4995,33 +5002,61 @@ fn cmd_git(ui: &mut Ui, command: &CommandHelper, args: &GitArgs) -> Result<(), C
     }
 }
 
-fn resolve_alias(ui: &mut Ui, args: Vec<String>) -> Vec<String> {
+fn string_list_from_config(value: config::Value) -> Option<Vec<String>> {
+    match value {
+        Value {
+            kind: config::ValueKind::Array(elements),
+            ..
+        } => {
+            let mut strings = vec![];
+            for arg in elements {
+                match arg {
+                    config::Value {
+                        kind: config::ValueKind::String(string_value),
+                        ..
+                    } => {
+                        strings.push(string_value);
+                    }
+                    _ => {
+                        return None;
+                    }
+                }
+            }
+            Some(strings)
+        }
+        _ => None,
+    }
+}
+
+fn resolve_alias(settings: &UserSettings, args: Vec<String>) -> Result<Vec<String>, CommandError> {
     if args.len() >= 2 {
         let command_name = args[1].clone();
-        if let Ok(alias_definition) = ui
-            .settings()
+        match settings
             .config()
-            .get_array(&format!("alias.{}", command_name))
+            .get::<config::Value>(&format!("alias.{}", command_name))
```

Comment:
This isn't a bug introduced by this patch, but `Config::get*()` may fail to "parse" the given `command_name`, and show cryptic error message.

```
% jj 'weird[]'  
Error: Config error: Digit
```

I couldn't find API to pass in parsed key (like `config.get(Path("alias", command_name))`) while working on the merge-tools settings, and ended up first extracting the whole table and look up without using the config API.